specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -165,6 +169,37 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
                 .then(async (accepted) =>
                   accepted === null ? taxon : getLastAccepted(accepted)
                 );
+
+            const collectionObject = determination.collection
+              ?.related as SpecifyResource<CollectionObject>;
+            collectionObject
+              .rgetPromise('collectionObjectType', true)
+              .then((coType: SpecifyResource<CollectionObjectType>) => {
+                /*
+                 * Have to set save blockers directly here to get this working.
+                 * Since following code has to wait for above rgetPromise to resolve, returning a Promise<BusinessRuleResult> for validation here is too slow and
+                 * does not get captured by business rules.
+                 */
+                if (
+                  idFromUrl(coType.get('taxonTreeDef')) ===
+                  idFromUrl(taxon?.get('definition') ?? '')
```

Comment:
(optional) 
`idFromUrl` is not needed here. 

The following should be equivalent, 

```suggestion
                  coType.get('taxonTreeDef') ===
                  (taxon?.get('definition') ?? '')
```